test/models/user_test.rb

```diff
@@ -157,15 +157,6 @@ def setup
     refute u.valid?
   end
 
-  test "mail is required for own user" do
-    user = FactoryBot.create(:user)
-    user.password = nil
-    # refute_valid user can check only one field and due to we need to set password to nil after adding current_password field to verify password change
-    as_user user do
-      refute_valid user, :mail
-    end
-  end
-
   test "hidden users don't need mail when updating" do
```

Comment:
This is a fair point. Currently, admins are not required to have email even if they have email enabled, as can be seen in `app/models/user.rb` on line 86:
```
:if => proc { |u| !AuthSourceHidden.where(:id => u.auth_source_id).any? && u.mail_enabled }
```
This behavior was already present before this PR. I am a little bit hesitant to change behavior around admin accounts as I am afraid it might break things that even the tests won't catch, but if we agree that this should be changed, I can of course do that.

Reply:
I agree we shouldn't change the behavior of admin accounts. However, I think this test should reflect the current behavior, where `mail_enabled` is no longer mandatory by default.

Reply:
Updated.